src/lavinmq/exchange/default.cr

```diff
@@ -6,22 +6,30 @@ module LavinMQ
       "direct"
     end
 
-    def bind(destination, routing_key, headers = nil)
-      raise "Access refused"
+    def bindings_details : Iterator(BindingDetails)
+      ([] of BindingDetails).each
     end
 
-    def unbind(destination, routing_key, headers = nil)
-      raise "Access refused"
+    def bindings : Iterator(Destination)
+      ([] of Destination).each
     end
 
-    def do_queue_matches(routing_key, headers = nil, & : Queue -> _)
+    protected def bindings(routing_key, headers) : Iterator(Destination)
       if q = @vhost.queues[routing_key]?
-        yield q
+        a = Array(Destination).new
+        a << q
+        a.each
```

Comment:
Use a Tuple to avoid allocations 
```suggestion
        {q}.each
```

Reply:
`{q}.each` gives a type error 

but `Slice(Destination).empty.each.chain({q}.each)` works

Reply:
What if you do `Tuple(Destination).new(q).each`?
Another way could be `Iterator(Destination).of(q).first(1)`, but this is probably two allocation while Tuple will be one. A third way is `Iterator(Destination).chain({ {q}.each })`, but also two allocations I think.